test/services/messengerSync/services/schoolSyncService.test.js

```diff
@@ -0,0 +1,99 @@
+const rabbitmqMock = require('../rabbitmqMock');
+
+const schoolSyncService = app.service('schools/:schoolId/messengerSync');
+
+describe('messenger schoolSync Service', () => {
+	let server;
+	before((done) => {
+		server = app.listen(0, done);
+	});
+
+	after((done) => {
+		server.close(done);
+		testObjects.cleanup();
+	});
+
+	describe('with rabbitMQ enabled', () => {
+		let configBefore;
+
+		before(async () => {
+			configBefore = Configuration.toObject(); // deep copy current config
+			Configuration.set('FEATURE_RABBITMQ_ENABLED', true);
+			mockery.enable({
+				warnOnReplace: false,
+				warnOnUnregistered: false,
+				useCleanCache: true,
+			});
+			mockery.registerMock('../../../utils/rabbitmq', rabbitmqMock);
+			delete require.cache[
+				require.resolve('../../../../src/services/messengerSync/services/schoolSyncService.js')
+			];
+			// eslint-disable-next-line global-require
+			const messengerSync = require('../../../../src/services/messengerSync');
+			app.configure(messengerSync);
+		});
+
+		after(async () => {
+			Configuration.parse(configBefore);
+		});
+
+		it('admin can trigger a schoolSync', async () => {
+			this.app = app;
+			const school = await testObjects.createTestSchool({ features: ['messenger'] });
+			const users = await Promise.all([
+				testObjects.createTestUser({ roles: ['administrator'], schoolId: school._id }),
+				testObjects.createTestUser({ roles: ['teacher'], schoolId: school._id }),
+				testObjects.createTestUser({ roles: ['student'], schoolId: school._id }),
+			]);
+			const params = await testObjects.generateRequestParamsFromUser(users[0]);
+			params.route = { schoolId: school._id.toString() };
+			await app.service('schools/:schoolId/messengerSync').create({}, params);
+
+			const testingQueue = rabbitmqMock.queues.matrix_sync_unpopulated;
+			expect(testingQueue).to.not.be.undefined;
+			expect(testingQueue.length).to.equal(3);
+
+			const firstMessage = JSON.parse(testingQueue[0]);
+			expect(users.map((u) => u._id.toString()).includes(firstMessage.userId)).to.be.true;
+			expect(firstMessage.schoolSync).to.be.true;
+
+			// todo: check rabbit Messages
+		});
+	});
+
+	describe('with rabbitMQ disabled', () => {
+		let configBefore;
+
+		before(async () => {
+			configBefore = Configuration.toObject(); // deep copy current config
+			Configuration.set('FEATURE_RABBITMQ_ENABLED', false);
+		});
+
+		after(async () => {
+			Configuration.parse(configBefore);
+		});
+
+		it('schoolSync responds with an error', async () => {
+			this.app = app;
+			const school = await testObjects.createTestSchool({ features: ['messenger'] });
+			const users = await Promise.all([
+				testObjects.createTestUser({ roles: ['administrator'], schoolId: school._id }),
+			]);
+			const params = await testObjects.generateRequestParamsFromUser(users[0]);
+			params.route = { schoolId: school._id.toString() };
+			try {
+				const result = await schoolSyncService.create({}, params);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['result' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5203703)